source/codegen/service_helpers.py

```diff
@@ -111,6 +111,8 @@ def _create_standard_arg(parameter):
         return f"&{parameter_name}_copy, "
     elif not is_array and is_output:
         return f"&{parameter_name}, "
+    elif not is_array and not is_output and "pointer" in parameter:
+        return f"&{parameter_name}, "
```

Comment:
@astarche this is the place where I tried to put a logic so that `&` is introduced in service implementation. But this is causing `&` to be introduced in some other drivers as well.

Reply:
Ah, I see. Some `pointer` fields are pointers on the stack that are passed by value:

```c++
auto reserved = nullptr;
library_->Calll(reserved);
```

And some, like this (and `is_size_param_passed_by_ptr`) we want to pass a pointer to a value on the stack.

```c++
auto timestamp = request->timestamp();
library_->Calll(&reserved);
```

Yeah, I like your proposal of adding a new flag. I would suggest that you still set "pointer" to true and add an additional flag like "input_passed_by_pointer" or "pass_address_to_driver" or something like that. Then you don't need to change the simple code that handles "pointer" for the library. And the new complexity/logic can go in this method and people reading it have some chance to understand how the decision making works.

And ... since we mentioned in the discussion. To implement this with an InputConverter. It would need to implement *implicit conversion to the pointer type*, NOT override the `&` operator (that was for OutputConverters). It would be pretty simple to implement though, if you prefer that option. That's how I would implement it, but everything is an InputConverter in nxsocket...

Reply:
I have implemented the 1st option. That was quite easy to implement.